torch_xla/csrc/elementwise.cpp

```diff
@@ -69,10 +69,16 @@ xla::XlaOp BuildRelu(xla::XlaOp input) {
                              0, input_shape.element_type(), input.builder()));
 }
 
-xla::XlaOp BuildHardshrink(xla::XlaOp input, const at::Scalar& lambda) {
+xla::XlaOp BuildHardshrink(xla::XlaOp input, xla::XlaOp lambda) {
```

Comment:
We need to be careful about these `at::Scalar` -> `xla::XlaOp` because currently the  default is f64. We can do a `MaybeCast` here, take a look at https://github.com/pytorch/xla/blame/f8b3dfd45d753a8844aca871cb39511022bb35ff/torch_xla/csrc/elementwise.cpp#L383 for example.

Reply:
If we don't cast, then we may get error
```
Seen floating point types of different precisions in %subtract.4 = f64[] subtract(f32[] %constant.3, f64[] %constant.1), metadata={op_type="aten__hardshrink" op_name="aten__hardshrink" source_file="__call__@core.py" source_line=1026}, but mixed precision is disallowed.
```
?

Also, shouldn't we cast the element type of xla::XlaOp to its original type of at::Scalar?

Reply:
check https://github.com/pytorch/xla/blob/76df13008a2ca84cc49cce8727f6c205a12b21d9/torch_xla/csrc/elementwise.cpp#L23 for existing behavior.  scalar often default to f64 when user don't specified it and will cause additional type promotion issue if we use its own type. It is easier to user the `other operand` type in this case.

Reply:
Got it. Thanks!

By "promotion issue", you meant when we have an operation with 2 mixed type operands, xla will try to convert one type to another implicitly, just like any c++ operator?

Reply:
yea.. we usually promote to "more complex" type like `f64` and `s64` which are slower to compute.